packages/core/src/components/overlay2/overlay2.tsx

```diff
@@ -307,6 +276,8 @@ export const Overlay2 = React.forwardRef<OverlayInstance, Overlay2Props>((props,
             }
         }
     }, [closeOverlay, getLastOpened, handleDocumentFocus, handleDocumentMousedown, id]);
+    const mostRecetOverlayWillClose = React.useRef(overlayWillClose);
```

Comment:
typo
```suggestion
    const mostRecentOverlayWillClose = React.useRef(overlayWillClose);
```

Reply:
ah thanks! I decided "mostRecent" was weird anyways and went with `overlayWillCloseRef` and also co-located it with the `useEffect` so it's a bit more clear why we have this ref